utils/src/config.rs

```diff
@@ -0,0 +1,713 @@
+    #[getset(get = "pub")]
+    cluster: ClusterConfig,
+    /// log configuration object
+    #[getset(get = "pub")]
+    log: LogConfig,
+    /// trace configuration object
+    #[getset(get = "pub")]
+    trace: TraceConfig,
+    /// auth configuration object
+    #[getset(get = "pub")]
+    auth: AuthConfig,
+}
+
+/// Duration Wrapper for meeting the orphan rule
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Deserialize)]
+pub struct ClusterDuration(Duration);
+
+impl FromStr for ClusterDuration {
+    type Err = anyhow::Error;
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.ends_with("us") {
+            if let Some(dur) = s.strip_suffix("us") {
+                Ok(ClusterDuration(Duration::from_micros(dur.parse()?)))
+            } else {
+                Err(anyhow!(format!("Failed to parse {s} to ClusterDuration")))
+            }
+        } else if s.ends_with("ms") {
+            if let Some(dur) = s.strip_suffix("ms") {
+                Ok(ClusterDuration(Duration::from_millis(dur.parse()?)))
+            } else {
+                Err(anyhow!(format!("Failed to parse {s} to ClusterDuration")))
+            }
+        } else if s.ends_with('s') {
+            if let Some(dur) = s.strip_suffix('s') {
+                Ok(ClusterDuration(Duration::from_secs(dur.parse()?)))
+            } else {
+                Err(anyhow!(format!("Failed to parse {s} to ClusterDuration")))
+            }
+        } else {
+            Err(anyhow!(format!("Invalid time unit:{s}")))
+        }
+    }
+}
+
+impl Deref for ClusterDuration {
+    type Target = Duration;
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Range Wrapper
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize)]
+pub struct ClusterRange(Range<u64>);
+
+impl FromStr for ClusterRange {
+    type Err = anyhow::Error;
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Some((start, end)) = s.split_once("..") {
+            Ok(ClusterRange(Range {
+                start: start.parse::<u64>()?,
+                end: end.parse::<u64>()?,
+            }))
+        } else {
+            Err(anyhow!(format!("Invalid cluster range:{s}")))
+        }
+    }
+}
+
+impl Deref for ClusterRange {
+    type Target = Range<u64>;
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// `ClusterDuration` deserialization formatter
+pub mod cluster_duration_format {
+    use super::ClusterDuration;
+    use serde::{self, Deserialize, Deserializer};
+    use std::str::FromStr;
+
+    /// deseralizes a cluster duration
+    #[allow(single_use_lifetimes)] //  the false positive case blocks us
+    pub(crate) fn deserialize<'de, D>(deserializer: D) -> Result<ClusterDuration, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        ClusterDuration::from_str(&s).map_err(serde::de::Error::custom)
+    }
```

Comment:
Why not just deserialize the string to Duration and return Duration here so that we can get rid of the `ClusterDuration` wrapper? I.e., move 34-54 lines here

Reply:
To satisfy the orphan rule in that we need to implement the `FromStr` trait here.

Reply:
We can implement a `parse_duration` for `Duration`, so it can be parsed by `clap`.  Doing so, we can remove the `ClusterDuration` wrapper